src/ralph/models/selector.py

```diff
@@ -152,3 +161,55 @@ def get_decision_tree(self, model_rules):
         false_subtree = self.get_decision_tree(false_subtree) if false_subtree else None
 
         return {root_rule: {True: true_subtree, False: false_subtree}}
+
+    def validate(self, input_file: TextIO, ignore_errors: bool, fail_on_unknown: bool):
+        """Validates JSON event strings line by line."""
+
+        count = 0
+        success_count = 0
+        for event_str in input_file:
+            try:
+                count += 1
+                yield self._validate_event(event_str)
+                success_count += 1
+            except (json.JSONDecodeError, TypeError) as err:
+                message = "Input event is not a valid JSON string!"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BackendException(message) from err
+            except UnknownEventException as err:
+                if fail_on_unknown:
+                    raise err
+            except ValidationError as err:
+                message = f"Input event is not a valid {err.model.__name__}!"
+                self._log_error(message, event_str, err)
+                if not ignore_errors:
+                    raise BackendException(message) from err
```

Comment:
This is a formatting issue right?

```suggestion
                    raise BadFormatException(message) from err
```

Reply:
You are right) This exception reflects better what happened)